### ISSUE TITLE
Support unknown search params

### ISSUE THREAD

**Issue by jamesagnew** [jamesagnew/hapi-fhir#176]
According to Grahame, it should be possible for a server to ignore search params it doesn't understand. The link-self in the resulting bundle should then be a link which only contains the search params that were actually used to respond to the search.

I'm thinking it might be nice to come up with some sort of priority system where methods which match exactly the given search params match first, then methods with optional params, then methods which don't match all of the params.

**Comment by wanghaisheng** [comment]
this would be strange.is there any other existing api or application in production use such mechanism

**Comment by jamesagnew** [comment]
I tend to agree, this is probably not something we're looking to do in our production systems. I'd like to at least add a flag though to allow it for those situations where people want it.

**Comment by chrisgrenz** [comment]
Would like to resurface this - at least make the "self" link reflect what was actually executed. Today, the _source parameter (meta.source) doesn't appear to be supported, but it fails to filter silently.
